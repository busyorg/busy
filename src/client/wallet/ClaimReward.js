import React from 'react';
import PropTypes from 'prop-types';
import {
  FormattedMessage,
  FormattedRelative,
  FormattedNumber,
  FormattedDate,
  FormattedTime,
} from 'react-intl';
import { Tooltip } from 'antd';
import formatter from '../helpers/steemitFormatter';

const getFormattedPayout = (
  rewardSteem,
  rewardSbd,
  rewardVests,
  totalVestingShares,
  totalVestingFundSteem,
) => {
  const payouts = [];
  const parsedRewardSteem = parseFloat(rewardSteem);
  const parsedRewardSbd = parseFloat(rewardSbd);
  const parsedRewardVests = parseFloat(
    formatter.vestToSteem(rewardVests, totalVestingShares, totalVestingFundSteem),
  );

  if (parsedRewardSteem > 0) {
    payouts.push(
      <span key="STEEM" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardSteem}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' STEEM'}
      </span>,
    );
  }

  if (parsedRewardSbd > 0) {
    payouts.push(
      <span key="SBD" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardSbd}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' SBD'}
      </span>,
    );
  }

  if (parsedRewardVests > 0) {
    payouts.push(
      <span key="SP" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardVests}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' SP'}
      </span>,
    );
  }

  return payouts;
};

const ClaimReward = ({
  timestamp,
  rewardSteem,
  rewardSbd,
  rewardVests,
  totalVestingShares,
  totalVestingFundSteem,
}) => (
  <div className="UserWalletTransactions__transaction">
    <div className="UserWalletTransactions__icon-container">
      <i className="iconfont icon-success_fill UserWalletTransactions__icon" />
    </div>
    <div className="UserWalletTransactions__content">
      <div className="UserWalletTransactions__content-recipient">
        <FormattedMessage id="claim_rewards" defaultMessage="Claim rewards" />
        <span className="UserWalletTransactions__payout">
          {getFormattedPayout(
            rewardSteem,
            rewardSbd,
            rewardVests,
            totalVestingShares,
            totalVestingFundSteem,
          )}
        </span>
      </div>
      <span className="UserWalletTransactions__timestamp">
        <Tooltip
          title={
            <span>
              <FormattedDate value={`${timestamp}Z`} /> <FormattedTime value={`${timestamp}Z`} />
            </span>
          }
        >
          <span>
            <FormattedRelative value={`${timestamp}Z`} />
          </span>
        </Tooltip>
      </span>
    </div>
  </div>
);

ClaimReward.propTypes = {
  timestamp: PropTypes.string.isRequired,
  rewardSteem: PropTypes.string.isRequired,
  rewardSbd: PropTypes.string.isRequired,
  rewardVests: PropTypes.string.isRequired,
  totalVestingShares: PropTypes.string.isRequired,
  totalVestingFundSteem: PropTypes.string.isRequired,
};

export default ClaimReward;
