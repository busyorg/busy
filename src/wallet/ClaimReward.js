import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative, FormattedNumber } from 'react-intl';

const getFormattedPayout = (rewardSteem, rewardSbd, rewardVests) => {
  const payouts = [];
  const parsedRewardSteem = parseFloat(rewardSteem);
  const parsedRewardSbd = parseFloat(rewardSbd);
  const parsedRewardVests = parseFloat(rewardVests);

  if (parsedRewardSteem > 0) {
    payouts.push(
      <span key="STEEM" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardSteem}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' STEEM'}
      </span>,
    );
  }

  if (parsedRewardSbd > 0) {
    payouts.push(
      <span key="SBD" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardSbd}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' SBD'}
      </span>,
    );
  }

  if (parsedRewardVests > 0) {
    payouts.push(
      <span key="SP" className="UserWalletTransactions__payout-rewards">
        <FormattedNumber
          value={parsedRewardVests}
          minimumFractionDigits={3}
          maximumFractionDigits={3}
        />
        {' SP'}
      </span>,
    );
  }

  return payouts;
};

const ClaimReward = ({ timestamp, rewardSteem, rewardSbd, rewardVests }) => (
  <div className="UserWalletTransactions__transaction">
    <div className="UserWalletTransactions__icon-container">
      <i className="iconfont icon-success_fill UserWalletTransactions__icon" />
    </div>
    <div className="UserWalletTransactions__content">
      <div className="UserWalletTransactions__content-recipient">
        <FormattedMessage id="claim_rewards" defaultMessage="Claim rewards" />
        <span className="UserWalletTransactions__payout">
          {getFormattedPayout(rewardSteem, rewardSbd, rewardVests)}
        </span>
      </div>
      <span className="UserWalletTransactions__timestamp">
        <FormattedRelative value={`${timestamp}Z`} />
      </span>
    </div>
  </div>
);

ClaimReward.propTypes = {
  timestamp: PropTypes.string.isRequired,
  rewardSteem: PropTypes.string.isRequired,
  rewardSbd: PropTypes.string.isRequired,
  rewardVests: PropTypes.string.isRequired,
};

export default ClaimReward;
