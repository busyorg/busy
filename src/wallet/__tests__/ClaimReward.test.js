import React from 'react';
import { shallow } from 'enzyme';
import ClaimReward from '../ClaimReward';

describe('(Component) ClaimReward', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        timestamp: '0',
        rewardSteem: '0 STEEM',
        rewardSbd: '0 SBD',
        rewardVests: '0 SP',
        totalVestingShares: '0',
        totalVestingFundSteem: '0',
      };
      const wrapper = shallow(<ClaimReward {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
