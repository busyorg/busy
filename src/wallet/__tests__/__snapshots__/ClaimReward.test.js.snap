// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ClaimReward with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="UserWalletTransactions__transaction"
  >
    <div
      className="UserWalletTransactions__icon-container"
    >
      <i
        className="iconfont icon-success_fill UserWalletTransactions__icon"
      />
    </div>
    <div
      className="UserWalletTransactions__content"
    >
      <div
        className="UserWalletTransactions__content-recipient"
      >
        <FormattedMessage
          defaultMessage="Claim rewards"
          id="claim_rewards"
          values={Object {}}
        />
        <span
          className="UserWalletTransactions__payout"
        />
      </div>
      <span
        className="UserWalletTransactions__timestamp"
      >
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          prefixCls="ant-tooltip"
          title={
            <span>
              <FormattedDate
                value="0Z"
              />
               
              <FormattedTime
                value="0Z"
              />
            </span>
          }
          transitionName="zoom-big-fast"
        >
          <span>
            <FormattedRelative
              updateInterval={10000}
              value="0Z"
            />
          </span>
        </Tooltip>
      </span>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="UserWalletTransactions__transaction"
    >
      <div
        className="UserWalletTransactions__icon-container"
      >
        <i
          className="iconfont icon-success_fill UserWalletTransactions__icon"
        />
      </div>
      <div
        className="UserWalletTransactions__content"
      >
        <div
          className="UserWalletTransactions__content-recipient"
        >
          <FormattedMessage
            defaultMessage="Claim rewards"
            id="claim_rewards"
            values={Object {}}
          />
          <span
            className="UserWalletTransactions__payout"
          />
        </div>
        <span
          className="UserWalletTransactions__timestamp"
        >
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            prefixCls="ant-tooltip"
            title={
              <span>
                <FormattedDate
                  value="0Z"
                />
                 
                <FormattedTime
                  value="0Z"
                />
              </span>
            }
            transitionName="zoom-big-fast"
          >
            <span>
              <FormattedRelative
                updateInterval={10000}
                value="0Z"
              />
            </span>
          </Tooltip>
        </span>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ClaimReward
        rewardSbd="0 SBD"
        rewardSteem="0 STEEM"
        rewardVests="0 SP"
        timestamp="0"
        totalVestingFundSteem="0"
        totalVestingShares="0"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "rewardSbd": "0 SBD",
          "rewardSteem": "0 STEEM",
          "rewardVests": "0 SP",
          "timestamp": "0",
          "totalVestingFundSteem": "0",
          "totalVestingShares": "0",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="UserWalletTransactions__transaction"
        >
          <div
            className="UserWalletTransactions__icon-container"
          >
            <i
              className="iconfont icon-success_fill UserWalletTransactions__icon"
            />
          </div>
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Claim rewards"
                id="claim_rewards"
                values={Object {}}
              />
              <span
                className="UserWalletTransactions__payout"
              />
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <Tooltip
                arrowPointAtCenter={false}
                autoAdjustOverflow={true}
                mouseEnterDelay={0.1}
                mouseLeaveDelay={0.1}
                placement="top"
                prefixCls="ant-tooltip"
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
                transitionName="zoom-big-fast"
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </Tooltip>
            </span>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="UserWalletTransactions__transaction"
        >
          <div
            className="UserWalletTransactions__icon-container"
          >
            <i
              className="iconfont icon-success_fill UserWalletTransactions__icon"
            />
          </div>
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Claim rewards"
                id="claim_rewards"
                values={Object {}}
              />
              <span
                className="UserWalletTransactions__payout"
              />
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <Tooltip
                arrowPointAtCenter={false}
                autoAdjustOverflow={true}
                mouseEnterDelay={0.1}
                mouseLeaveDelay={0.1}
                placement="top"
                prefixCls="ant-tooltip"
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
                transitionName="zoom-big-fast"
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </Tooltip>
            </span>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ClaimReward
    rewardSbd="0 SBD"
    rewardSteem="0 STEEM"
    rewardVests="0 SP"
    timestamp="0"
    totalVestingFundSteem="0"
    totalVestingShares="0"
  />,
}
`;
